src/Compilers/CSharp/Test/Emit/Emit/EditAndContinue/EditAndContinueTests.cs

```diff
@@ -7183,5 +7183,920 @@ static void F()
         }
-    }
+
+        [Fact]
+        public void OutVar()
+        {
+            var source0 = MarkedSource(@"
+class C
+{
+    static void F(out int x, out int y) { x = 1; y = 2; }
+    static int G() { F(out int <N:0>x</N:0>, out var <N:1>y</N:1>); return x + y; }
+}");
+            var source1 = MarkedSource(@"
+class C
+{
+    static void F(out int x, out int y) { x = 1; y = 2; }
+    static int G() { F(out int <N:0>x</N:0>, out var <N:1>z</N:1>); return x + z; }
+}");
+
+            var source2 = MarkedSource(@"
+class C
+{
+    static void F(out int x, out int y) { x = 1; y = 2; }
+    static int G() { F(out int <N:0>x</N:0>, out int <N:1>y</N:1>); return x + y; }
+}");
+
+            var compilation0 = CreateStandardCompilation(source0.Tree, options: ComSafeDebugDll);
+            var compilation1 = compilation0.WithSource(source1.Tree);
+            var compilation2 = compilation1.WithSource(source2.Tree);
+
+            var f0 = compilation0.GetMember<MethodSymbol>("C.G");
+            var f1 = compilation1.GetMember<MethodSymbol>("C.G");
+            var f2 = compilation2.GetMember<MethodSymbol>("C.G");
+
+            var v0 = CompileAndVerify(compilation0);
+            v0.VerifyIL("C.G", @"
+{
+  // Code size       19 (0x13)
+  .maxstack  2
+  .locals init (int V_0, //x
+                int V_1, //y
+                int V_2)
+  IL_0000:  nop
+  IL_0001:  ldloca.s   V_0
+  IL_0003:  ldloca.s   V_1
+  IL_0005:  call       ""void C.F(out int, out int)""
+  IL_000a:  nop
+  IL_000b:  ldloc.0
+  IL_000c:  ldloc.1
+  IL_000d:  add
+  IL_000e:  stloc.2
+  IL_000f:  br.s       IL_0011
+  IL_0011:  ldloc.2
+  IL_0012:  ret
+}
+");
+
+            var md0 = ModuleMetadata.CreateFromImage(v0.EmittedAssemblyData);
+            var generation0 = EmitBaseline.CreateInitialBaseline(md0, v0.CreateSymReader().GetEncMethodDebugInfo);
+            var diff1 = compilation1.EmitDifference(
+                generation0,
+                ImmutableArray.Create(
+                    new SemanticEdit(SemanticEditKind.Update, f0, f1, preserveLocalVariables: true)));
+
+            diff1.VerifyIL("C.G", @"
+{
+  // Code size       19 (0x13)
+  .maxstack  2
+  .locals init (int V_0, //x
+                int V_1, //z
+                int V_2)
+  IL_0000:  nop
+  IL_0001:  ldloca.s   V_0
+  IL_0003:  ldloca.s   V_1
+  IL_0005:  call       ""void C.F(out int, out int)""
+  IL_000a:  nop
+  IL_000b:  ldloc.0
+  IL_000c:  ldloc.1
+  IL_000d:  add
+  IL_000e:  stloc.2
+  IL_000f:  br.s       IL_0011
+  IL_0011:  ldloc.2
+  IL_0012:  ret
+}
+");
+            var v1 = CompileAndVerify(compilation1);
+            var md1 = ModuleMetadata.CreateFromImage(v1.EmittedAssemblyData);
+
+            var generation1 = EmitBaseline.CreateInitialBaseline(md1, v1.CreateSymReader().GetEncMethodDebugInfo);
+            var diff2 = compilation2.EmitDifference(
+                generation1,
```

Comment:
>generation1 [](start = 16, length = 11)

This is testing the first generation again. We need to test 2nd generation and instead pass in diff1.NextGeneration.